<?xml version="1.0"?>
<launch>
  <arg name="sim" default="true"/>
  <arg name="rviz" default="true"/>
  <arg name="ats" default="false"/>
  <arg name="robot_name" default="perseverance"/>
  <arg name="joint_control" default="false"/>
  <arg name="use_namespace" default="false"/>
  <arg name="x" default="0"/>
  <arg name="y" default="0"/>
  <arg name="z" default="0"/>    
  <arg name="yaw" default="0"/>

  <!-- Time -->
  <param name="use_sim_time" value="true" unless="$(arg sim)"/>
  <param name="use_sim_time" value="false" if="$(arg sim)"/>

  <include file="$(find m2020_urdf_models)/launch/perseverance_config.launch">
    <arg name="sim" value="$(arg sim)"/>
    <arg name="robot_name" value="$(arg robot_name)"/>
    <arg name="use_namespace" value="$(arg use_namespace)"/>
  </include>
  
  <!-- Planner Node -->
  <include file="$(find m2020_urdf_models)/launch/perseverance_planner_node.launch">
    <arg name="sim" value="$(arg sim)"/>
    <arg name="robot_name" value="$(arg robot_name)"/>
    <arg name="use_namespace" value="$(arg use_namespace)"/>
  </include>

  <!-- IC -->
  <include file="$(find m2020_urdf_models)/launch/perseverance_controls.launch">
    <arg name="ats" value="$(arg ats)"/>
    <arg name="robot_name" value="$(arg robot_name)"/>
    <arg name="joint_control" value="$(arg joint_control)"/>
    <arg name="use_namespace" value="$(arg use_namespace)"/>
  </include>

  <!-- Navigation -->
  <include file="$(find craftsman_navigation_tools)/launch/craftsman_navigation_setup.launch">
    <arg name="robot_name" value="$(arg robot_name)"/>
    <arg name="use_namespace" value="$(arg use_namespace)"/>
    <arg name="rviz_sim" value="$(arg sim)"/>
    <arg name="nav_controller" value="mobile_base_controller/cmd_vel" if="$(arg sim)"/>
    <arg name="nav_controller" value="base_controller/command" unless="$(arg sim)"/>
    <arg name="nav_interface" value="basic_navigators::PlanarNavInterface"/>
    <arg name="nav_executor" value="basic_navigation_executors::Path2DPIDExecutor"/>
    <arg name="jnt_cmd" value="joint_state_controller/joint_command"/>
    <arg name="nav_path_step" value="0.015"/>
    <arg name="nav_file_path" value="$(find m2020_urdf_models)/config/perseverance/perseverance_navigation.yaml"/>
    
    <arg name="x" value="$(arg x)"/>
    <arg name="y" value="$(arg y)"/>
    <arg name="z" value="$(arg z)"/>    
    <arg name="yaw" value="$(arg yaw)"/>
    
  </include>


  <!-- Display -->
<!--  <group if="$(arg use_namespace)">
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find m2020_urdf_models)/rviz/perseverance_ns.rviz" respawn="false" output="screen" if="$(arg rviz)"/>
  </group>
-->
  <group unless="$(arg use_namespace)">
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find m2020_urdf_models)/rviz/perseverance.rviz" respawn="false" output="screen" if="$(arg rviz)"/>
  </group>

</launch>
